zerver/lib/send_email.py

```diff
@@ -119,7 +122,12 @@ class EmailNotDeliveredException(Exception):
 def send_email(template_prefix: str, to_user_ids: Optional[List[int]]=None,
                to_emails: Optional[List[str]]=None, from_name: Optional[str]=None,
                from_address: Optional[str]=None, reply_to_email: Optional[str]=None,
-               language: Optional[str]=None, context: Dict[str, Any]={}) -> None:
+               language: Optional[str]=None, context: Dict[str, Any]={},
+               sender: Optional[str]=None) -> None:
+    if sender == FromAddress.tokenized_no_reply_placeholder:
+        from_address = FromAddress.tokenized_no_reply_address()
+    if sender == FromAddress.no_reply_placeholder:
+        from_address = FromAddress.NOREPLY
```

Comment:
I don't think we need a `sender` parameter; just put the value in the existing `from_address` parameter, and move this logic to sit next to this code (above):

```
    if from_address is None:
        from_address = FromAddress.NOREPLY
```

Reply:
@timabbott  Do we have to use placeholder everywhere where NOREPLY  or tokenized_no_reply is present or we have to confine it only for places where future_send is used as I think we only require it for Schedule objects. Also do we need placeholder for SUPPORT too?